envoy/network/dns.h

```diff
@@ -56,7 +56,7 @@ class ActiveDnsQuery {
   virtual void addTrace(uint8_t trace) PURE;
 
   /** Return the DNS query traces. */
-  virtual OptRef<const std::vector<Trace>> getTraces() PURE;
+  virtual absl::optional<std::string> getTraces() PURE;
```

Comment:
so I meant string not optional string.  I'm fine with this but I think you could simplify further by just returning en empty string in the "no trace" case

Reply:
Done.